From: Mike Hommey <mh+mozilla@glandium.org>
Date: Mon, 5 Aug 2013 17:01:29 +0900
Subject: Bug 901200 - Change SharedStub definition in tier-3 xptcstubs to
 better fit reality and make the compiler happy now that nsresult is an enum

---
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf32.cpp           | 2 +-
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf64.cpp           | 2 +-
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp            | 2 +-
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_netbsd.cpp    | 2 +-
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_openbsd.cpp   | 2 +-
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_solaris.cpp   | 2 +-
 xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparcv9_solaris.cpp | 2 +-
 7 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf32.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf32.cpp
index f4c5826..5b70cc6 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf32.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf32.cpp
@@ -126,7 +126,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint32_t methodIndex,
   return result;
 }
 
-extern "C" int SharedStub(uint64_t,uint64_t,uint64_t,uint64_t,
+extern "C" nsresult SharedStub(uint64_t,uint64_t,uint64_t,uint64_t,
  uint64_t,uint64_t,uint64_t,uint64_t,uint64_t,uint64_t *);
 
 /* Variable a0-a7 were put there so we can have access to the 8 input
diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf64.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf64.cpp
index 2e0add3..b58a94b 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf64.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_ipf64.cpp
@@ -129,7 +129,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint32_t methodIndex,
   return result;
 }
 
-extern "C" int SharedStub(uint64_t,uint64_t,uint64_t,uint64_t,
+extern "C" nsresult SharedStub(uint64_t,uint64_t,uint64_t,uint64_t,
  uint64_t,uint64_t,uint64_t,uint64_t,uint64_t,uint64_t *);
 
 /* Variable a0-a7 were put there so we can have access to the 8 input
diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp
index 116d690..94650f7 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_pa32.cpp
@@ -124,7 +124,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint32_t methodIndex,
   return result;
 }
 
-extern "C" int SharedStub(int);
+extern "C" nsresult SharedStub(int);
 
 #define STUB_ENTRY(n)       \
 nsresult nsXPTCStubBase::Stub##n()  \
diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_netbsd.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_netbsd.cpp
index 1f4ad01..3b6b01f 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_netbsd.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_netbsd.cpp
@@ -93,7 +93,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint32_t methodIndex, uint32_t* args)
     return result;
 }
 
-extern "C" int SharedStub(int, int*);
+extern "C" nsresult SharedStub(int, int*);
 
 #define STUB_ENTRY(n) \
 nsresult nsXPTCStubBase::Stub##n() \
diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_openbsd.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_openbsd.cpp
index 212f7b8..456a8b0 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_openbsd.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_openbsd.cpp
@@ -96,7 +96,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint32_t methodIndex, uint32_t* args)
     return result;
 }
 
-extern "C" int SharedStub(int, int*);
+extern "C" nsresult SharedStub(int, int*);
 
 #define STUB_ENTRY(n) \
 nsresult nsXPTCStubBase::Stub##n() \
diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_solaris.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_solaris.cpp
index 424f936..2569932 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_solaris.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparc_solaris.cpp
@@ -88,7 +88,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint32_t methodIndex, uint32_t* args)
     return result;
 }
 
-extern "C" int SharedStub(int, int*);
+extern "C" nsresult SharedStub(int, int*);
 
 #define STUB_ENTRY(n) \
 nsresult nsXPTCStubBase::Stub##n() \
diff --git a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparcv9_solaris.cpp b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparcv9_solaris.cpp
index 5078a6f..e97c2de 100644
--- a/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparcv9_solaris.cpp
+++ b/xpcom/reflect/xptcall/src/md/unix/xptcstubs_sparcv9_solaris.cpp
@@ -80,7 +80,7 @@ PrepareAndDispatch(nsXPTCStubBase* self, uint64_t methodIndex, uint64_t* args)
     return result;
 }
 
-extern "C" int SharedStub(int, int*);
+extern "C" nsresult SharedStub(int, int*);
 
 #define STUB_ENTRY(n) \
 nsresult nsXPTCStubBase::Stub##n() \
